ledger-tool/src/main.rs

```diff
@@ -3676,17 +3689,13 @@ fn main() {
             ("purge", Some(arg_matches)) => {
                 let start_slot = value_t_or_exit!(arg_matches, "start_slot", Slot);
                 let end_slot = value_t!(arg_matches, "end_slot", Slot).ok();
-                let no_compaction = arg_matches.is_present("no_compaction");
```

Comment:
it'd be good to keep this arg check around, but only emit a deprecation warning when it's passed.  i'm guessing that's not how it went into master though, so maybe add back in a followup there and backport separately

Reply:
Addressed in master in: https://github.com/solana-labs/solana/pull/31178

Thoughts on landing that in master, and then cherry-picking into this branch. I guess it makes things slightly more complicated, but I kind of like the aspect of "cleaning things up" on the stable branch.

Reply:
could just as easily merge separately.  there's value in the 1:1::master:backport mapping

Reply:
Done